frontend/src/service/defenceService.ts

```diff
@@ -63,60 +65,61 @@ async function configureDefence(
 	return response.status === 200;
 }
 
-async function resetDefenceConfig(
-	defenceId: string,
-	configId: string
+async function resetDefenceConfigItem(
+	defenceId: DEFENCE_ID,
+	configItemId: DEFENCE_CONFIG_ITEM_ID,
+	level: LEVEL_NAMES
 ): Promise<DefenceResetResponse> {
 	const response = await sendRequest(`${PATH}resetConfig`, {
 		method: 'POST',
 		headers: {
 			'Content-Type': 'application/json',
 		},
-		body: JSON.stringify({ defenceId, configId }),
+		body: JSON.stringify({ defenceId, configItemId, level }),
 	});
 	return (await response.json()) as DefenceResetResponse;
 }
 
-function validatePositiveNumberConfig(config: string) {
+function validatePositiveNumberConfig(value: string) {
 	// config is a number greater than zero
-	return !isNaN(Number(config)) && Number(config) > 0;
+	return !isNaN(Number(value)) && Number(value) > 0;
 }
 
-function validateNonEmptyStringConfig(config: string) {
+function validateNonEmptyStringConfig(value: string) {
 	// config is non empty string and is not a number
-	return config !== '' && !Number(config);
+	return value !== '' && !Number(value);
```

Comment:
Unrelated to this issue, but does this strike you as odd? We are validating a string is non-empty, but we are also disallowing numeric-only content such as "123456". And yet something like "123 456" or "123,456" or "123_456" _is_ allowed cos the result will be `NaN`.

I'll add it to my todo list to investigate.

Reply:
good spot. Would be a nice thing to fix. Though you could argue that a nonsense numerical config (say, system prompt) is still valid even if it won't do much good.

Reply:
I was about to change this too, and then I thought I could add tests, but I stopped myself before this PR bloats even more